src/inference/src/threading/ie_istreams_executor.cpp

```diff
@@ -241,6 +332,75 @@ Parameter IStreamsExecutor::Config::GetConfig(const std::string& key) const {
+                                                         const int num_small_cores) {
+    const auto base_streams_total = num_big_cores_phys * 2 + num_small_cores;
+    const auto num_big_cores = num_big_cores_phys * 2;
+
+    config._small_core_offset = num_big_cores_phys * 2;
+    if (config._threads) {
+        // limit threads to reasonable range
+        auto threads = config._threads > base_streams_total ? base_streams_total : config._threads;
+        // threads are placed on big cores under ideal condition
+        if (threads <= num_big_cores) {
+            // First, ensure the balanced use of physical core and logical core
+            config._big_core_streams = std::max(1, config._streams / 2 * 2);
+            config._threads_per_stream_big = std::max(1, threads / config._streams);
+            // The remaining threads are placed on the small core.
+            config._threads_per_stream_small =
+                std::min(const_cast<int&>(num_small_cores), threads - config._threads_per_stream_big * config._big_core_streams);
+            config._small_core_streams = config._threads_per_stream_small == 0 ? 0 : 1;
+        } else {
+            // big cores first, then small cores
+            // big_streams = all_streams * big_cores / base_streams
+            int big_core_streams = config._streams * num_big_cores % base_streams_total == 0
+                                       ? config._streams * num_big_cores / base_streams_total
+                                       : (config._streams * num_big_cores + base_streams_total) / base_streams_total;
+            // _big_core_streams must be even
+            config._big_core_streams = std::min(std::max(1, config._streams / 2 * 2), (big_core_streams + 1) / 2 * 2);
+            config._threads_per_stream_big = std::max(1, num_big_cores / config._big_core_streams);
+            config._small_core_streams = config._streams - config._big_core_streams;
+            if (config._small_core_streams > 0) {
+                // Normally, threads_per_stream_small = threads_per_stream_big, but special conditions are as
+                // follows: Pcore=8, Ecore=4, custom set nthreads=17, nstreams=9, calculate result: big_streams=8,
+                // small_streams=1. There are only 1 threads left, threads_per_stream_small can only be 1.
+                const int remain_threads = threads - config._threads_per_stream_big * config._big_core_streams;
+                config._threads_per_stream_small =
+                    std::min(remain_threads / config._small_core_streams, config._threads_per_stream_big);
+            }
+        }
+    } else {  // user not set nthreads
+        // Ensure the balanced use of physical core and logical core, _threads_per_stream_big must be divisible and
+        // smaller than num_big_cores_phys.
+        int threads_per_stream_big = std::min(num_big_cores_phys, std::max(1, base_streams_total / config._streams));
```

Comment:
threads_per_stream_big can be odd.

For example on 6P+4E platform, if customer nstreams is 5, threads_per_stream_big and threads_per_stream_small are 3. 5 streams include 4 Pcore streams and 1 Ecore streams.

Reply:
Yes, it is modified.